stories/__tests__/SQFormCheckboxGroup.stories.test.js

```diff
@@ -0,0 +1,194 @@
+        SHOPPING_LIST_OPTIONS[0].label
+      );
+      expect(checkboxLabel).not.toBeChecked();
+
+      userEvent.click(checkboxLabel);
+      expect(checkboxLabel).toBeChecked();
+    });
+
+    it('should uncheck when checkbox is checked and user clicks', () => {
+      render(<SQFormCheckboxGroup size="auto" />);
+
+      const checkboxLabel = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[1].label
+      );
+      expect(checkboxLabel).not.toBeChecked();
+
+      userEvent.click(checkboxLabel);
+      expect(checkboxLabel).toBeChecked();
+
+      userEvent.click(checkboxLabel);
+      expect(checkboxLabel).not.toBeChecked();
+    });
+
+    it('should render as required when isRequired is true', () => {
+      render(<SQFormCheckboxGroup size="auto" isRequired />);
+
+      const required = screen.getByText('Required');
+      expect(required).toBeInTheDocument();
+    });
+
+    it('should render an extra checkbox when adding select all capability', () => {
+      render(<SQFormCheckboxGroup size="auto" shouldUseSelectAll />);
+
+      const selectAllCheckbox = screen.getByLabelText('All');
+      expect(selectAllCheckbox).toBeInTheDocument();
+      expect(selectAllCheckbox).not.toBeChecked();
+    });
+
+    it('should check all checkboxes when select all checkbox checked', () => {
+      render(<SQFormCheckboxGroup size="auto" shouldUseSelectAll />);
+
+      const selectAllCheckbox = screen.getByLabelText('All');
+      const cb1 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[0].label);
+      const cb2 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[1].label);
+      const cb3 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[2].label);
+      const cb4 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[3].label);
+      const cb5 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[4].label);
+      expect(selectAllCheckbox).not.toBeChecked();
+      expect(cb1).not.toBeChecked();
+      expect(cb2).not.toBeChecked();
+      expect(cb3).not.toBeChecked();
+      expect(cb4).not.toBeChecked();
+      expect(cb5).not.toBeChecked();
+
+      userEvent.click(selectAllCheckbox);
+
+      expect(selectAllCheckbox).toBeChecked();
+      expect(cb1).toBeChecked();
+      expect(cb2).toBeChecked();
+      expect(cb3).toBeChecked();
+      expect(cb4).toBeChecked();
+      expect(cb5).toBeChecked();
+    });
+
+    it('should uncheck all checkboxes when select all checkbox is unchecked', () => {
+      render(<SQFormCheckboxGroup size="auto" shouldUseSelectAll />);
+
+      const selectAllCheckbox = screen.getByLabelText('All');
+      const cb1 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[0].label);
+      const cb2 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[1].label);
+      const cb3 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[2].label);
+      const cb4 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[3].label);
+      const cb5 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[4].label);
+
+      userEvent.click(selectAllCheckbox);
```

Comment:
I thnk it'd be better if we used initialValues to set the checked state for the test instead of checking them in the test.

We have a test else where that the initialValues is doing what we need so we can use it here with confidence. That way we don't have to check that each checkbox is checked in this test since this particular test doesn't really care about that. Instead it should only care about unchecking the select all.

Reply:
I have attempted this, but the select all does not seem to play nice with `initalValues`. I can't find a way to set it. I checked the html and it's showing no value (`value >`) so I think that's probably why.

Reply:
Hm, not sure I follow, sorry. Although if you've found an issue then it's surely one we'll want to fix in the component.

You can set the select all in initialValues by using `selectAll_[groupNameHere]: true`